File: AWS/AmazonTranslate.cs
// dotnet add package AWSSDK.Translate --version 3.3.101.7
using System;
using System.Threading.Tasks;
using Amazon;
using Amazon.Translate;
using Amazon.Translate.Model;

namespace TranslateSample
{
    class Program
    {
        static async Task Main()
        {
            var region = RegionEndpoint.APNortheast1;
            var client = new AmazonTranslateClient(region);
            var request = new TranslateTextRequest
            {
                SourceLanguageCode = "ja",
                TargetLanguageCode = "en",
                Text = "おはようございます"
            };
            var resp = await client.TranslateTextAsync(request);
            var result = resp.TranslatedText;
            Console.WriteLine(result);
        }
    }
}

File: AWS/amazon-translate.js
(async () => {

    // 参考: https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Translate.html

    // AWS SDK をインポートする
    // npm install aws-sdk
    var AWS = require('aws-sdk');
    AWS.config.update({ region: 'ap-northeast-1' });

    // テキストを翻訳して、結果を画面に表示する
    var translate = new AWS.Translate();
    var result = await translate.translateText({
        SourceLanguageCode: 'en',
        TargetLanguageCode: 'ja',
        Text: 'This is a pen.',
    }).promise();
    console.log(result);

})();

